crates/bevy_pbr/src/meshlet/persistent_buffer_impls.rs

```diff
@@ -0,0 +1,65 @@
+use super::{persistent_buffer::PersistentGpuBufferable, Meshlet, MeshletBoundingSphere};
+use std::sync::Arc;
+
+impl PersistentGpuBufferable for Arc<[u8]> {
+    type Metadata = ();
+
+    fn size_in_bytes(&self) -> u64 {
+        self.len() as u64
+    }
+
+    fn write_bytes_le(&self, _: Self::Metadata, buffer: &mut Vec<u8>) {
+        buffer.extend_from_slice(self);
+    }
+}
+
+impl PersistentGpuBufferable for Arc<[u32]> {
+    type Metadata = u64;
+
+    fn size_in_bytes(&self) -> u64 {
+        self.len() as u64 * 4
+    }
+
+    fn write_bytes_le(&self, offset: Self::Metadata, buffer: &mut Vec<u8>) {
+        let offset = offset as u32 / 48;
+
+        for index in self.iter() {
+            let bytes = (*index + offset).to_le_bytes();
+            buffer.extend_from_slice(&bytes);
+        }
+    }
+}
+
+impl PersistentGpuBufferable for Arc<[Meshlet]> {
+    type Metadata = (u64, u64);
+
+    fn size_in_bytes(&self) -> u64 {
+        self.len() as u64 * 12
```

Comment:
need docs on the 12, 48, and 16 in this file

Reply:
Or perhaps `std::mem::size_of::<T>()`?